autorest/adal/cli.go

```diff
@@ -47,13 +50,42 @@ func AzureCLIProfilePath() (string, error) {
 }
 
 // ToToken converts an AzureCLIToken to a Token
-func (t AzureCLIToken) ToToken() Token {
-	return Token{
+func (t AzureCLIToken) ToToken() (*Token, error) {
+	tokenExpirationDate, err := ParseAzureCLIExpirationDate(t.ExpiresOn)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing Token Expiration Date %q: %+v", t.ExpiresOn, err)
+	}
+
+	difference := tokenExpirationDate.Sub(expirationBase)
+	seconds := difference.Seconds()
+
+	token := Token{
 		AccessToken:  t.AccessToken,
 		Type:         t.TokenType,
-		ExpiresIn:    strconv.Itoa(t.ExpiresIn),
-		ExpiresOn:    t.ExpiresOn,
+		ExpiresIn:    "3600",
+		ExpiresOn:    strconv.Itoa(int(seconds)),
 		RefreshToken: t.RefreshToken,
 		Resource:     t.Resource,
 	}
+	return &token, nil
+}
+
+// ParseAzureCLIExpirationDate parses either a Azure CLI or CloudShell date into a time object
+func ParseAzureCLIExpirationDate(input string) (*time.Time, error) {
+	log.Printf("[DEBUG] Token Date: %s", input)
```

Comment:
Let's remove this line before we commit this.

Reply:
Updated :)